src/TwigComponent/src/Twig/ComponentNode.php

```diff
@@ -43,6 +43,16 @@ public function compile(Compiler $compiler): void
     {
         $compiler->addDebugInfo($this);
 
+        // since twig/twig 3.9.0: Using the internal "twig_to_array" function is deprecated.
+        $compiler
+            ->write('$toArray = function ($data) {')
+            ->write('if (method_exists(Twig\Extension\CoreExtension::class, \'toArray\')) {')
```

Comment:
I think this test should be done on compile time and not on render time... Or we will call method_exists a billion time per render..

So i'd make something like 

if (method_exists(...))  {
    $compiler->write('return Twig\Extension\CoreExtension::toArray($data);')
} else {
    $compiler->write('return twig_to_array'...)
}

Reply:
That's a nice idea 👍